frontend/src/components/RegistrationForm/RegistrationForm.js

```diff
@@ -0,0 +1,235 @@
+import { 
+  Button,
+  CardContent,
+  IconButton,
+  InputAdornment,
+  Grid,
+  Paper,
+  TextField,
+  Typography
+} from '@material-ui/core';
+import visibilityIcon from '../../assets/generic/visibility.svg';
+import visibilityOffIcon from '../../assets/generic/visibility-off.svg';
+import productIcon from "../../assets/generic/product-icon.svg"
+import { emailRegex, passwordRegex, nameRegex } from "../../constants/regex"
+
+export default function RegistrationForm(props) {
+  const [firstName, setFirstName] = useState("")
+  const [lastName, setLastName] = useState("")
+  const [email, setEmail] = useState("")
+  const [password, setPassword] = useState("")
+  const [confirmPassword, setConfirmPassword] = useState("")
+  const [showPassword, setShowPassword] = useState(false)
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false)
+  const [showError, setShowError] = useState({ 
+    errorFirstName: false,
+    errorLastName: false,
+    errorEmail: false,
+    errorPassword: false,
+    errorConfirmPassword: false
+  })
+
+  const handleRegister = (event) => {
+    event.preventDefault()
+    if (!nameRegex.test(firstName)) {
+      setShowError({ ...showError, errorFirstName: true })
+      return
+    }
+    if (!nameRegex.test(lastName)) {
+      setShowError({ ...showError, errorLastName: true })
+      return
+    }
+    if (!emailRegex.test(email)) {
+      setShowError({ ...showError, errorEmail: true })
+      return
+    }
+    if (!passwordRegex.test(password)) {
+      setShowError({ ...showError, errorPassword: true })
+      return
+    }
+    if (confirmPassword !== password) {
+      setShowError({ ...showError, errorConfirmPassword: true })
+      return
+    }
+    // [TM]: TODO WDB-28
+    props.onRegister(firstName, lastName, email, password)
+  }
+
+  const handleFirstNameChange = (event) => {
+    setFirstName(event.target.value)
+    setShowError({ ...showError, errorFirstName: false })
+  }
+
+  const handleLastNameChange = (event) => {
+    setLastName(event.target.value)
+    setShowError({ ...showError, errorLastName: false })
+  }
+
+  const handleEmailChange = (event) => {
+    setEmail(event.target.value)
+    setShowError({ ...showError, errorEmail: false })
+  }
+
+  const handlePasswordChange = (event) => {
+    setPassword(event.target.value)
+    setShowError({ ...showError, errorPassword: false })
+  }
+
+  const handleConfirmPasswordChange = (event) => {
+    setConfirmPassword(event.target.value)
+    setShowError({ ...showError, errorConfirmPassword: false })
+  }
+
+  const handleClickShowPassword = () => setShowPassword(!showPassword)
+  const handleClickShowConfirmPassword = () => setShowConfirmPassword(!showConfirmPassword)
+  const handleMouseDownPassword = (event) => event.preventDefault()
+
+  return (
+    <Paper elevation={3}>
+      <CardContent
+        direction="column"
+        align="center"
+        justify="center"
+      >
+        <img
+          src={productIcon}
```

Comment:
is this the same as in LoginPage? If so, can you just create a separate reusable component to be used here, please?

Reply:
Don't see a reason here, project timeline is short, I think we should not spend time for tiny details as this